components/Filters/RemoveFilter.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { c } from 'ttag';
+import {
+    Button,
+    ConfirmModal,
+    Alert,
+    useApiWithoutResult,
+    useModals,
+    useNotifications,
+    useEventManager
+} from 'react-components';
+import { deleteFilter } from 'proton-shared/lib/api/filters';
+import { noop } from 'proton-shared/lib/helpers/function';
+
+function RemoveFilter({ filter, className, onRemoveFilter }) {
+    const { call } = useEventManager();
+    const { createModal } = useModals();
+    const { createNotification } = useNotifications();
+    const { request, loading } = useApiWithoutResult(deleteFilter);
+
+    const handleConfirmConfirmModal = async () => {
+        await request(filter.ID);
+        call();
```

Comment:
If you are not using `onRemoveFilter` when you use this component you should wait on the event manager to refresh the filter collection before displaying the success notification.
```suggestion
        await call();
```

Reply:
Also if the API call for the event manager fails, the user will still see the success notification and the collection will not be updated on the UI.